### PR TITLE
Pr7318 mariadb104 compatibility

### PR DESCRIPTION
**Pull request by grooverdan**
change to a stored procedure to delete users.

closes #7318
closes #7398

related to: https://github.com/vitessio/vitess/issues/5362

### CODE REVIEW

File: docker/bootstrap/Dockerfile.mariadb104 [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    && add-apt-repository 'deb [arch=amd64] http://ftp.osuosl.org/pub/mariadb/repo/10.4/debian buster main' \
+    && apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+       mariadb-server-10.4 \
+       libmariadb-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# Bootstrap Vitess
+WORKDIR /vt/src/vitess.io/vitess
+
+ENV MYSQL_FLAVOR MariaDB103
```
Comment (an3l) [inline docker/bootstrap/Dockerfile.mariadb104:17]: `104`

File: docker/lite/Dockerfile.mariadb104 [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Copyright 2019 The Vitess Authors.
```
Comment (systay) [inline docker/lite/Dockerfile.mariadb104:1]: nit: 2021

**Comment by askdba** [comment]
A preliminary test is showing the cluster is up partially with MariaDB 10.4. Further debugging needed.   

```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa25366]

goroutine 1 [running]:
vitess.io/vitess/go/exit.doRecover(0xaf2aa0, 0x1223ba0, 0x1223b00)
	/vt/src/vitess.io/vitess/go/exit/exit.go:92 +0x192
vitess.io/vitess/go/exit.Recover()
	/vt/src/vitess.io/vitess/go/exit/exit.go:69 +0x48
panic(0xaf2aa0, 0x1223ba0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
vitess.io/vitess/go/cmd/vtctldclient/internal/command.commandInitShardPrimary(0x122e700, 0xc000162480, 0x2, 0x3, 0x0, 0x0)
	/vt/src/vitess.io/vitess/go/cmd/vtctldclient/internal/command/reparents.go:64 +0x246
github.com/spf13/cobra.(*Command).execute(0x122e700, 0xc0000300b0, 0x3, 0x3, 0x122e700, 0xc0000300b0)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x122d200, 0xc000030080, 0x0, 0x6)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
main.main()
	/vt/src/vitess.io/vitess/go/cmd/vtctldclient/main.go:41 +0x126
I0208 17:14:32.797698    1038 main.go:67] I0208 17:14:32.796534 tablet_executor.go:240] Received DDL request. strategy=direct
W0208 17:14:42.813299    1038 main.go:67] W0208 17:14:42.810906 schema.go:107] Failed to reload schema on replica tablet zone1-0000000101 in commerce/0 (use vtctl ReloadSchema to try again): rpc error: code = DeadlineExceeded desc = context deadline exceeded
W0208 17:14:42.815138    1038 main.go:67] W0208 17:14:42.812223 schema.go:107] Failed to reload schema on replica tablet zone1-0000000102 in commerce/0 (use vtctl ReloadSchema to try again): rpc error: code = DeadlineExceeded desc = context deadline exceeded
I0208 17:14:42.816477    1038 main.go:67] I0208 17:14:42.812333 tablet_executor.go:240] Received DDL request. strategy=direct
W0208 17:14:52.826274    1038 main.go:67] W0208 17:14:52.825214 schema.go:107] Failed to reload schema on replica tablet zone1-0000000102 in commerce/0 (use vtctl ReloadSchema to try again): rpc error: code = DeadlineExceeded desc = context deadline exceeded
W0208 17:14:52.828620    1038 main.go:67] W0208 17:14:52.825281 schema.go:107] Failed to reload schema on replica tablet zone1-0000000101 in commerce/0 (use vtctl ReloadSchema to try again): rpc error: code = DeadlineExceeded desc = context deadline exceeded
I0208 17:14:52.829970    1038 main.go:67] I0208 17:14:52.825358 tablet_executor.go:240] Received DDL request. strategy=direct
W0208 17:15:02.840997    1038 main.go:67] W0208 17:15:02.840511 schema.go:107] Failed to reload schema on replica tablet zone1-0000000102 in commerce/0 (use vtctl ReloadSchema to try again): rpc error: code = DeadlineExceeded desc = context deadline exceeded
W0208 17:15:02.844796    1038 main.go:67] W0208 17:15:02.840608 schema.go:107] Failed to reload schema on replica tablet zone1-0000000101 in commerce/0 (use vtctl ReloadSchema to try again): rpc error: code = DeadlineExceeded desc = context deadline exceeded
New VSchema object:
{
  "tables": {
    "corder": {

    },
    "customer": {

    },
    "product": {

    }
  }
}
If this is not what you expected, check the input data (as JSON parsing will skip unexpected fields).
Waiting for vtgate to be up...
vtgate is up!
Access vtgate at http://1cab37271d75:15001/debug/status
vitess@1cab37271d75:/vt/local$ mysql
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MySQL connection id is 1
Server version: 5.7.9-Vitess mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MySQL [(none)]> show vitess_tablets;
+-------+----------+-------+------------+-------------+------------------+--------------+----------------------+
| Cell  | Keyspace | Shard | TabletType | State       | Alias            | Hostname     | MasterTermStartTime  |
+-------+----------+-------+------------+-------------+------------------+--------------+----------------------+
| zone1 | commerce | 0     | MASTER     | SERVING     | zone1-0000000100 | 1cab37271d75 | 2021-02-08T17:14:02Z |
| zone1 | commerce | 0     | REPLICA    | NOT_SERVING | zone1-0000000101 | 1cab37271d75 |                      |
| zone1 | commerce | 0     | RDONLY     | NOT_SERVING | zone1-0000000102 | 1cab37271d75 |                      |
+-------+----------+-------+------------+-------------+------------------+--------------+----------------------+
3 rows in set (0.001 sec)

MySQL [(none)]>
MySQL [(none)]> \s
--------------
mysql  Ver 15.1 Distrib 10.4.17-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

Connection id:		1
Current database:
Current user:		vt_app@localhost
SSL:			Not in use
Current pager:		stdout
Using outfile:		''
Using delimiter:	;
Server:			MySQL
Server version:		5.7.9-Vitess mariadb.org binary distribution
Protocol version:	10
Connection:		127.0.0.1 via TCP/IP
Server characterset:	utf8
Db     characterset:	utf8
Client characterset:	utf8
Conn.  characterset:	utf8
TCP port:		15306
--------------
```

**Comment by askdba** [comment]
Further testing and we're failing to set up a replica ;
```ERROR: logging before flag.Parse: E0209 15:02:36.098164     981 syslogger.go:149] can't connect to syslog
Error: rpc error: code = Unknown desc = tablet zone1-0000000102 InitReplica failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded;tablet zone1-0000000101 InitReplica failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded
Usage:
   InitShardPrimary [flags]

Flags:
      --force                            will force the reparent even if the provided tablet is not a master or the shard master
  -h, --help                             help for InitShardPrimary
      --wait-replicas-timeout duration   time to wait for replicas to catch up in reparenting (default 30s)
```

We suspect it's something to do with setting GTID Current_Pos vs Slave_Pos

```vitess@dc61117c3a90:/vt/local$ mysql --socket=/vt/vtdataroot/vt_0000000101/mysql.sock -h localhost -u vt_dba
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 47
Server version: 10.4.17-MariaDB-1:10.4.17+maria~buster-log mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]> show slave status\G
*************************** 1. row ***************************
                Slave_IO_State: Waiting for master to send event
                   Master_Host: dc61117c3a90
                   Master_User: vt_repl
                   Master_Port: 17100
                 Connect_Retry: 10
               Master_Log_File: vt-0000000100-bin.000001
           Read_Master_Log_Pos: 5401
                Relay_Log_File: vt-0000000101-relay-bin.000002
                 Relay_Log_Pos: 986
         Relay_Master_Log_File: vt-0000000100-bin.000001
              Slave_IO_Running: Yes
             Slave_SQL_Running: No
               Replicate_Do_DB:
           Replicate_Ignore_DB:
            Replicate_Do_Table:
        Replicate_Ignore_Table:
       Replicate_Wild_Do_Table:
   Replicate_Wild_Ignore_Table:
                    Last_Errno: 1146
                    Last_Error: Error executing row event: 'Table '_vt.reparent_journal' doesn't exist'
                  Skip_Counter: 0
           Exec_Master_Log_Pos: 1154
               Relay_Log_Space: 5550
               Until_Condition: None
                Until_Log_File:
                 Until_Log_Pos: 0
            Master_SSL_Allowed: No
            Master_SSL_CA_File:
            Master_SSL_CA_Path:
               Master_SSL_Cert:
             Master_SSL_Cipher:
                Master_SSL_Key:
         Seconds_Behind_Master: NULL
 Master_SSL_Verify_Server_Cert: No
                 Last_IO_Errno: 0
                 Last_IO_Error:
                Last_SQL_Errno: 1146
                Last_SQL_Error: Error executing row event: 'Table '_vt.reparent_journal' doesn't exist'
   Replicate_Ignore_Server_Ids:
              Master_Server_Id: 589375127
                Master_SSL_Crl:
            Master_SSL_Crlpath:
                    Using_Gtid: Slave_Pos
                   Gtid_IO_Pos: 0-589375127-18
       Replicate_Do_Domain_Ids:
   Replicate_Ignore_Domain_Ids:
                 Parallel_Mode: conservative
                     SQL_Delay: 0
           SQL_Remaining_Delay: NULL
       Slave_SQL_Running_State:
              Slave_DDL_Groups: 2
Slave_Non_Transactional_Groups: 0
    Slave_Transactional_Groups: 1
1 row in set (0.002 sec)

MariaDB [(none)]> show master status;
+--------------------------+----------+--------------+------------------+
| File                     | Position | Binlog_Do_DB | Binlog_Ignore_DB |
+--------------------------+----------+--------------+------------------+
| vt-0000000101-bin.000001 |      650 |              |                  |
+--------------------------+----------+--------------+------------------+
1 row in set (0.001 sec)
```

**Pull request closed without merging**